test/test_nn.py

```diff
@@ -7223,6 +7223,30 @@ def test_cosine_embedding_loss_margin_no_reduce(self):
                          loss_reference_fns['CosineEmbeddingLoss'](input1, input2, target,
                                                                    margin=0.5, reduction='none'))
 
+    def test_cosine_embedding_loss_with_diff_type(self):
+        for device in device_():
+            for dt1 in torch.testing.get_all_math_dtypes(device):
```

Comment:
how long does this test take?  It's like...1000 combinations?

Reply:
It takes about 1 sec

Reply:
Update: it takes around 300ms